src/tools/perf/perftest_rte.h

```diff
@@ -0,0 +1,60 @@
+/**
+* Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2024. ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#ifndef UCX_PERFTEST_RTE_H
+#define UCX_PERFTEST_RTE_H
+
+#include <ucs/type/status.h>
+
+
+typedef enum {
+    PERFTEST_RTE_SOCK = 0,
+    PERFTEST_RTE_MPI,
+    PERFTEST_RTE_MAD,
+    PERFTEST_RTE_MAX
+} perftest_rte_type_t;
+
+struct perftest_context;
+
+typedef struct perftest_rte {
+    perftest_rte_type_t type;
+    struct perftest_rte *next;
```

Comment:
I'd suggest to use ucs/list here: it's used in similar places, can reuse list macros, and list_del is simpler